pkg/apis/autoscaling/v1alpha1/pa_types.go

```diff
@@ -80,6 +80,9 @@ type PodAutoscalerSpec struct {
 	// is responsible for quickly right-sizing.
 	ScaleTargetRef corev1.ObjectReference `json:"scaleTargetRef"`
 
+	// Reachable specifies whether or not the `ScaleTargetRef` can be reached.
```

Comment:
A quick explanation on why this is a tri-state would be great. The meaning of `nil` especially.

Reply:
+1 to a comment about this.  Generally, it's for the upgrade path (where folks have a mix of [un]reachable PAs that don't have this set, where we need to be paranoid).

Unclear whether we'd use this in other circumstances (e.g. initial readiness) yet.